lib/services/ios-watch-app-service.ts

```diff
@@ -0,0 +1,84 @@
+	constructor(protected $fs: IFileSystem,
+		protected $pbxprojDomXcode: IPbxprojDomXcode,
+		protected $xcode: IXcode) {
+			super($fs, $pbxprojDomXcode, $xcode);
+	}
+
+	public async addWatchAppFromPath({watchAppFolderPath, projectData, platformData, pbxProjPath}: IAddWatchAppFromPathOptions): Promise<boolean> {
+		const targetUuids: string[] = [];
+		const appPath = path.join(watchAppFolderPath, IOS_WATCHAPP_FOLDER);
+		const extensionPath = path.join(watchAppFolderPath, IOS_WATCHAPP_EXTENSION_FOLDER);
+
+		if (!this.$fs.exists(appPath) || !this.$fs.exists(extensionPath)) {
+			return false;
+		}
+
+		const appFolder = this.getTargetDirectories(appPath)[0];
+		const extensionFolder = this.getTargetDirectories(extensionPath)[0];
+
+		const project = new this.$xcode.project(pbxProjPath);
+		project.parseSync();
+
+		const watchApptarget = this.addTargetToProject(appPath, appFolder, IOSNativeTargetTypes.watchApp, project, platformData, project.getFirstTarget().uuid);
+		this.configureTarget(
+			appFolder,
+			path.join(appPath, appFolder),
+			`${projectData.projectIdentifiers.ios}.${IOSWatchAppService.WATCH_APP_IDENTIFIER}`,
+			"watchapp.json",
+			watchApptarget,
+			project
+		);
+		targetUuids.push(watchApptarget.uuid);
+
+		const watchExtensionTarget = this.addTargetToProject(extensionPath, extensionFolder, IOSNativeTargetTypes.watchExtension, project, platformData, watchApptarget.uuid);
+		this.configureTarget(
+			extensionFolder,
+			path.join(extensionPath, extensionFolder),
+			`${projectData.projectIdentifiers.ios}.${IOSWatchAppService.WATCH_APP_IDENTIFIER}.${IOSWatchAppService.WACTCH_EXTENSION_IDENTIFIER}`,
+			"extension.json",
+			watchExtensionTarget,
+			project);
+		targetUuids.push(watchExtensionTarget.uuid);
+
+		this.$fs.writeFile(pbxProjPath, project.writeSync({omitEmptyValues: true}));
+		this.prepareSigning(targetUuids, projectData, pbxProjPath);
+
+		return true;
+	}
+
+	public removeWatchApp({pbxProjPath}: IRemoveWatchAppOptions): void {
+		const project = new this.$xcode.project(pbxProjPath);
+		project.parseSync();
+		project.removeTargetsByProductType(IOSNativeTargetProductTypes.watchApp);
+		project.removeTargetsByProductType(IOSNativeTargetProductTypes.watchExtension);
+		this.$fs.writeFile(pbxProjPath, project.writeSync({omitEmptyValues: true}));
+	}
+
+	private configureTarget(targetName: string, targetPath: string, identifier: string, configurationFileName: string, target: IXcode.target, project: IXcode.project) {
+		const targetConfigurationJsonPath = path.join(targetPath, configurationFileName);
+
+		const identifierParts = identifier.split(".");
+		identifierParts.pop();
+		const wkAppBundleIdentifier = identifierParts.join(".");
+
+		this.setXcodeTargetBuildConfigurationProperties([
+			{name: "PRODUCT_BUNDLE_IDENTIFIER", value: identifier},
+			{name: "SDKROOT", value: "watchos"},
+			{name: "TARGETED_DEVICE_FAMILY", value: IOSDeviceTargets.watchos},
+			{name: "WATCHOS_DEPLOYMENT_TARGET", value: 5.2},
```

Comment:
should we hardcode this?

Reply:
I agree that its not obvious, but I left a way for the user to override this setting in the `watchapp.json` and `extension.json`. The method call [here ](https://github.com/NativeScript/nativescript-cli/pull/4568/files#diff-6a6e27b65c72d3aa7c4350bd602c3a1cR79) will read the JSON file and set all target build configuration properties listed in the property - the code is [here](https://github.com/NativeScript/nativescript-cli/pull/4568/files#diff-609e22cb6bb3cad20d7ba450784b4184R89). 

For an example how the JSON file will look like you can check point 7 in the related issue description [here](https://github.com/NativeScript/nativescript-cli/issues/4541).

Reply:
Thanks for the great explanation and please excuse me for not checking the related issue.